server/src/prisma/schema.prisma

```diff
@@ -0,0 +1,252 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+//  ███    ███  ██████  ██████  ███████ ██      ███████
+//  ████  ████ ██    ██ ██   ██ ██      ██      ██
+//  ██ ████ ██ ██    ██ ██   ██ █████   ██      ███████
+//  ██  ██  ██ ██    ██ ██   ██ ██      ██           ██
+//  ██      ██  ██████  ██████  ███████ ███████ ███████
+
+/// TODO: replicate mongoose static methods
+/// TODO: check prisma capability to regex validate type and id_ fields
```

Comment:
we can handle this regex thing with zod if we need to, don't worry too much about this one

Reply:
Oh alright